common/RequestsCleaner.js

```diff
@@ -0,0 +1,46 @@
+class RequestsCleaner {
+	/**
+	 * @param {*} calls
+	 * @param {number} intervalMs
+	 * @param {number} timeoutMs
+	 */
+	constructor(calls, intervalMs, timeoutMs) {
+		this.calls = calls;
+		this.intervalMs = intervalMs;
+		this.timeoutMs = timeoutMs;
+		this.cleanInterval = null;
+	}
+
+	start() {
+		if (this.cleanInterval) {
+			throw new Error('Requests cleaner is already running');
+		}
+		this.cleanInterval = setInterval(this.verifySignatures, this.intervalMs);
+	}
+
+	stop() {
+		clearInterval(this.cleanInterval);
+		this.cleanInterval = null;
+	}
+
+	/**
+	 * @param {string} signature
+	 * @param {number} timeout
+	 * @returns {boolean}
+	 */
+	static isOutdated(signature, timeout) {
```

Comment:
Do you really want to pass the timeout every time? Will it ever not be `this.timeoutMs`?